### ISSUE TITLE
SSL validation fails on servers with SNI

### ISSUE THREAD

**Issue by WhyNotHugo** [geier/pycarddav#58]
When more than one SSL/TLS domain is hosted on a same IPv4 host, SNI support is required. It is not unusual to host several domains on a single host.

Without SNI support, there are only two ways to host several domains on a single host:
- Purchase a large IPv4 block, and use one-per-domain (extremely expensive).
- Make sure all your clients have IPv6 (not all location where I roam have IPv6 connectivity).

In short, SSL/TSL support is mostly broken as-is.
